### DISCUSSION TITLE
Passing Jacobian (Dfun) to Minimizer.leastsq() when dependent parameters are present

### DISCUSSION THREAD

**Discussion by Andrea-gm** [lmfit/lmfit-py#966] [Q&A]
I would like to pass a Jacobian function to Minimiser.leastsq(), having a lot of parameters that are independent, and some that are dependent.
10 years ago (see https://github.com/lmfit/lmfit-py/issues/42), it has been stated that the dimension of the Jacobian should equal the number of independent parameters. I'm wondering whether this is still the case, since I could not find information in the docs about this, or if a different implementation has been introduced since then. Thanks.

**Comment by newville** [comment]
@Andrea-gm There is an example of using `dfun` at https://github.com/lmfit/lmfit-py/blob/master/examples/example_fit_with_derivfunc.py
See also https://github.com/lmfit/lmfit-py/blob/master/tests/test_bounded_jacobian.py

The code outlined in that very old, closed issue was not ever added to the code base. 

I believe that Parameter bounds are handled.  And I believe that if you have any fixed or constrained parameters, your `Dfun` function should return 0 for those.

But we don't have very good examples or tests for Dfun with mixtures of "freely varying Parameters" and "constrained values".   This is considered to be an "expert option", and you may have to test things carefully.    If you have any examples or suggestions, this part of the code could probably use some attention.

**Reply by Andrea-gm** [reply]
Thanks a lot for the quick reply. I'l try what you suggested and let you know if it works.
